### PR TITLE
[Jetsurvey] Implementing the redesign of the survey screen (part 1)

### PR DESCRIPTION
**Pull request by florina-muntenescu**
* Changing the progress indicator, the previous and next buttons and the question text.
* The next button is now enabled only when an answer was selected
* Capitalising button texts throughout the app.

### CODE REVIEW

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyQuestions.kt [hunk 1]
```diff
@@ -58,11 +85,20 @@ fun Question(
...
         Spacer(modifier = Modifier.preferredHeight(44.dp))
-        Text(
-            text = stringResource(id = question.questionText),
-            style = MaterialTheme.typography.h4,
-            modifier = Modifier.fillMaxWidth()
-        )
+        Row(
+            modifier = Modifier.fillMaxWidth().background(
+                color = MaterialTheme.colors.questionBackground,
+                shape = RoundedCornerShape(12.dp)
```
Comment (nickbutcher) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyQuestions.kt:91]: Use a theme shape?
```suggestion
                shape = MaterialTheme.shapes.small
```

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyScreen.kt [hunk 2]
```diff
@@ -139,31 +145,46 @@ private fun SurveyResult(result: SurveyState.Result, modifier: Modifier = Modifi
...
 @Composable
 private fun SurveyTopAppBar(
-    @StringRes surveyTitle: Int,
+    questionIndex: Int,
+    totalQuestionsCount: Int,
     onBackPressed: () -> Unit
 ) {
-    TopAppBar(
-        title = {
+    Column(modifier = Modifier.fillMaxWidth()) {
```
Comment (nickbutcher) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyScreen.kt:155]: If using ConstraintLayout here, I'd lay out the progress indicator as well and remove the Column

File: Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/theme/Theme.kt [hunk 3]
```diff
@@ -56,6 +56,14 @@ val DarkThemeColors = darkColors(
 val Colors.snackbarAction: Color
     get() = if (isLight) Purple300 else Purple700
 
+@Composable
```
Comment (nickbutcher) [inline Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/theme/Theme.kt:59]: These color extensions aren't ideal, can they be derived from theme colors like `onSurface`? Perhaps push back on design?

Comment (florina-muntenescu) [reply Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/theme/Theme.kt:59]: Will do. I'll change them when I hear back.

**Review by nickbutcher — APPROVED** [review]

**Pull request merged**